test/redix/stream/consumer_test.exs

```diff
@@ -48,6 +51,36 @@ defmodule Redix.Stream.ConsumerTest do
     assert_receive {:streamed, ^stream_name, _id, %{"key_1" => "value_1"}}, 5_000
   end
 
+  @tag :integration
+  test "it should connect and stream a single message and ack async", %{
+    cmd_connection: cmd_connection,
+    stream_name: stream_name
+  } do
+    {:ok, redix_1} = Redix.start_link()
+    pid = self()
+
+    {:ok, consumer} =
+      Consumer.start_link(
+        redix_1,
+        stream_name,
+        fn stream, "test", id, values ->
+          send(pid, {:streamed, stream, id, values})
+          :defer
+        end,
+        group_name: "test",
+        consumer_name: "consumer1"
+      )
+
+    # allow consumer time to connect
+    :timer.sleep(500)
+
+    {:ok, _msg_id} = Redix.Stream.produce(cmd_connection, stream_name, %{"key_1" => "value_1"})
+
+    assert_receive {:streamed, ^stream_name, id, %{"key_1" => "value_1"}}, 5_000
+
+    assert Consumer.ack(consumer, id) == 1
+  end
```

Comment:
Do we want to do any kind of check to see the message was acked? Even just checking to see if we booted up a new stream it wouldn't show up again versus if we didn't ack, it would? Might be too complicated, but would be cool to see in action.

Reply:
This was acking 0 until I fixed an issue and started acking 1, so I think its checked, but could do more probably (though I think covered in other units)

Reply:
Ok, added a second consumer and it gets the second message and not the first